src/components/MainPageModel.ts

```diff
@@ -0,0 +1,282 @@
+            products.sort((item1, item2) => item2.price - item1.price);
+        }
+        if (sortString === 'RatingASC') {
+            products.sort((item1, item2) => item1.rating - item2.rating);
+        }
+        if (sortString === 'RatingDESC') {
+            products.sort((item1, item2) => item2.rating - item1.rating);
+        }
+        if (sortString === 'DiscountASC') {
+            products.sort((item1, item2) => item1.discountPercentage - item2.discountPercentage);
+        }
+        if (sortString === 'DiscountDESC') {
+            products.sort((item1, item2) => item2.discountPercentage - item1.discountPercentage);
+        }
+        return products;
+    }
+
+    filterPrice(products: IProduct[], minPrice: number, maxPrice: number) {
+        if (minPrice === this.minPriceProducts && maxPrice === this.maxPriceProducts) return products;
+        const productsFiltered = this.products.filter((cur) => {
+            return cur.price >= minPrice && cur.price <= maxPrice;
+        });
+        return productsFiltered;
+    }
+
+    filterStock(products: IProduct[], minPrice: number, maxPrice: number) {
+        if (minPrice === this.minStockProducts && maxPrice === this.maxStockProducts) return products;
+        const productsFiltered = this.products.filter((cur) => {
+            return cur.stock >= minPrice && cur.stock <= maxPrice;
+        });
+        return productsFiltered;
+    }
+
+    getProductsToShow(products: IProduct[], filter: IFilterData) {
+        const productFilteredByCategory: IProduct[] = this.filterByCategory(products, filter.categories);
+        const productFilteredByBrand: IProduct[] = this.filterByBrand(products, filter.brands);
+        const productFilteredBySearch: IProduct[] = this.filterBySearch(products, filter.search);
+        const productFilteredByPrice: IProduct[] = this.filterPrice(products, filter.minPrice, filter.maxPrice);
+        const productFilteredByStock: IProduct[] = this.filterStock(products, filter.minStock, filter.maxStock);
+        const productFiltered: IProduct[] = [];
+        productFilteredByCategory.map((itemCategory) => {
+            productFilteredByBrand.map((itemBrand) => {
+                if (itemBrand === itemCategory) {
+                    productFilteredBySearch.map((itemSearch) => {
+                        if (itemBrand === itemSearch) {
+                            productFilteredByPrice.map((itemPrice) => {
+                                if (itemSearch === itemPrice) {
+                                    productFilteredByStock.map((itemStock) => {
+                                        if (itemStock === itemPrice) {
+                                            productFiltered.push(itemStock);
+                                        }
+                                    });
+                                }
+                            });
+                        }
+                    });
+                }
+            });
+        });
```

Comment:
Use `filter` method here instead of `map`

Reply:
Done -> a1d44dc8e64c1e3a3e3ac3f9a53d724b8aebb41f